virtcontainers/kata_agent.go

```diff
@@ -1003,6 +1054,14 @@ func (k *kataAgent) createContainer(sandbox *Sandbox, c *Container) (p *Process,
 
 	k.handleShm(grpcSpec, sandbox)
 
+	caps := sandbox.hypervisor.capabilities()
+	if !caps.is9pSupported() {
```

Comment:
I would do this check and the file copy from `c.mountSharedDirMounts()` instead. This means we centralize everything in the `container.go` implementation. But that also means that you need to define `copyFile()` as part of the agent interface, which makes sense to me as you're asking to copy a file to the guest through the agent support.